### ISSUE TITLE
Batch Gradient Descent, SGD, and Mini-batch confusion

### ISSUE THREAD

**Issue by rafahuaman** [udacity/dlnd-issue-reports#152]

**Comment by rafahuaman** [comment]
Video 22 in the "Intro to Tensorflow" lessons, refers to an SGD with mini-batch algorithm as simply SGD. This was confusing to me.

I had to seek information elsewhere in order to make sure I was not misunderstanding. Eventually, I found a [video](https://www.youtube.com/watch?v=UfNU3Vhv5CA) which explained the differences between Batch and Stochastic Gradient Descent.  This video combined with the following information from the [SCD Wikipedia article](https://en.wikipedia.org/wiki/Stochastic_gradient_descent) gave me the full understanding 
> A compromise between computing the true gradient and the gradient at a single example, is to compute the gradient against more than one training example (called a "mini-batch") at each step.
